src/interface/comm.rs

```diff
@@ -205,6 +205,10 @@ impl Socket {
         unsafe {libc::listen(self.raw_sys_fd, backlog)}
     }
 
+    pub fn ioctl(&self, request: u32, argptr: *mut i32) -> i32 {
```

Comment:
I'm not sure we should call to libc ioctl? We can just set it non-blocking with fcntl? @jesings

Reply:
I think because we can do it with fcntl and we're not supporting any other ioctls for now we should do it with fcntl.

Reply:
Updated this. Any comments on the new implementation?

Reply:
I think it would be better if you directly called our emulated Socket fcntl call from ioctl, because that way it's clearer if we ever add more ioctl flags that that won't work right.